scalafix-core/shared/src/main/scala/scalafix/internal/rule/RuleCtxImpl.scala

```diff
@@ -37,6 +37,7 @@ case class RuleCtxImpl(tree: Tree, config: ScalafixConfig) extends RuleCtx {
   lazy val comments: AssociatedComments = AssociatedComments(tokens)
   lazy val input: Input = tokens.head.input
   lazy val escapeHatch = EscapeHatch(tree, comments)
+  lazy val diffDisable = config.diffs.map(diffs => new DiffDisable(diffs))
```

Comment:
If possible I think we should try to keep this out of RuleCtxImpl

Reply:
If the cli uses a different Rule.apply here https://github.com/scalacenter/scalafix/blob/bb0edd04fa59c291e4174f3900314b5f56e37e91/scalafix-cli/src/main/scala/scalafix/cli/CliRunner.scala#L125 then we can remove GitDiff out of ScalafixConfig and RuleCtx. What do you think?

Reply:
I'm not too sure what you mean here.

Reply:
In CliRunner we are currently using the convenience method `Rule.apply(RuleCtx)` that runs ctx.printLintMessage`. However, if we do that by ourselves here then we can filter the messages in the cli without adding new private methods to `RuleCtx`. Does that make sense? `printLintMessage` is already a hack that should ideally be removed from `RuleCtx`, we should avoid adding new methods like that.